topics/tutorial-first-steps-with-kotlin-rpc.topic

```diff
@@ -0,0 +1,600 @@
+    </chapter>
+    <chapter title="What is RPC?" id="what-is-rpc">
+        <chapter title="Local vs. remote procedure calls" id="local-vs-remote">
+            <p>
+                Anyone with programming experience will be familiar with the concept of a procedure call. This is a
+                fundamental concept in any programming language. Technically, these are
+                local procedure calls, since they always take place within the same program.
+            </p>
+            <p>
+                A remote procedure call is when the function call and parameters are in some way transferred over a
+                network, so that the implementation can occur within a separate VM/executable. The return value travels
+                the opposite path back to the machine where the invocation was made.
+            </p>
+            <p>
+                It is easiest to think of the machine where the invocation occurred as the client and the machine where
+                the implementation resides as the server. This does not necessarily have to be the case, however. RPC
+                calls could occur in both directions, as part of a peer architecture. But to keep things simple, let’s
+                assume a client/server deployment.
+            </p>
+        </chapter>
+        <chapter title="RPC framework fundamentals" id="rpc-framework-fundamentals">
+            <p>
+                Certain fundamentals must be provided by any RPC framework. They are inevitable when implementing remote
+                procedure calls within a conventional IT infrastructure. The terminology can vary, and responsibilities
+                can be divided in different ways, but every RPC framework must provide:
+            </p>
+            <list style="decimal">
+                <li>
+                    A way to declare the procedures that will be invoked remotely. In object-oriented programming, an
+                    interface is the logical choice. This could be the interface construct provided by the current
+                    language or some kind of language-neutral standard, such as <a
+                        href="https://webidl.spec.whatwg.org/">the Web IDL used by the W3C</a>
+                </li>
+                <li>
+                    A means to specify the types used for parameters and return values. Once again, you could use a
+                    language-neutral standard. However, it may be simpler to annotate standard data type declarations in
+                    the current language.
+                </li>
+                <li>
+                    Helper classes that will be used to convert the procedure invocation into a format that can be sent
+                    over the network and unpack the return value that results. These are often referred to as
+                    <format style="italic">client stubs</format>
+                    . They may be created at build time or dynamically at runtime.
```

Comment:
I think we can make this a bit clearer. From the first read, it is not obvious, who is the stub - the serialized procedure or the one who serializes. While the latter is true, it may not be so understandable for the first-time reader, who is not well into the RPC business

Reply:
Reworded this a bit. Let me know if this is enough or if there's a link with more information we can include.

Reply:
Yeah, that sound good. I would just replace `build time` with `compile time`, wdyt?